test/conftest.py

```
@@ -37,3 +43,107 @@
+            10000, 65535
+        )  # Choose a port in the dynamic/private range
+        if is_port_free(random_port):
+            break
+    else:
+        raise Exception("Could not find a free port after 10 attempts")
+
+    rtsp_address = f"rtsp://localhost:{random_port}/test"
+
+    # Create go2rtc.yaml dynamically
+    config_path = tmp_path_factory.mktemp("go2rtc") / "go2rtc.yaml"
+    with open(config_path, "w") as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)